src/editor/tools/tool.h

```diff
@@ -78,17 +89,18 @@ class EditorTool {
 	virtual bool has_size_one() const {
 		return false;
 	}
-	virtual EditorActionArgs format_args_impl(EditorInteractive& parent) {
-		return EditorActionArgs(parent);
-	}
-	virtual int32_t handle_click_impl(const Widelands::NodeAndTriangle<>&,
-	                                  EditorInteractive&,
-	                                  EditorActionArgs*,
-	                                  Widelands::Map*) = 0;
-	virtual int32_t handle_undo_impl(const Widelands::NodeAndTriangle<>&,
-	                                 EditorInteractive&,
-	                                 EditorActionArgs*,
-	                                 Widelands::Map*) {
+	virtual EditorActionArgs format_args_impl() {
+		return EditorActionArgs(parent_);
+	}
+
+	virtual std::string format_conf_description_impl(const ToolConf&) {
+		return "";
+	}
```

Comment:
This is so you don't have to implement the method in every tool. Many of the tools don't have any settings to store to the history. I think it would be annoying to have to add the method in every tool when they don't really need it. I could make it an interface.